### PR TITLE
add languages field in scala3-book.

### LINKED ISSUE #2559: Fix front matter of zh-cn translation of scala 3 book
e.g. https://docs.scala-lang.org/zh-cn/overviews/scala3-book/why-scala-3.html

without the front matter, the pages all have the wrong url, and have no layout applied

Edit: they still need the `language` field in order to have the language dropdown

### PR DESCRIPTION
**Pull request by benluo**
fixes #2559

### CODE REVIEW

File: _zh-cn/overviews/scala3-book/first-look-at-types.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 title: 类型初探
 type: chapter
 description: This page provides a brief introduction to Scala's built-in data types, including Int, Double, String, Long, Any, AnyRef, Nothing, and Null.
+languages:[en, ru]
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/first-look-at-types.md:5]: the `languages` field should only go on the English translation, the `zh-cn` pages should have
```yaml
language: zh-cn
```

File: _zh-cn/overviews/scala3-book/fp-functions-are-values.md [hunk 2]
```diff
@@ -91,12 +92,18 @@ fruits.map(_.toUpperCase)       // List(APPLE, PEAR)
 fruits.flatMap(_.toUpperCase)   // List(A, P, P, L, E, P, E, A, R)
 
 val nums = List(5, 1, 3, 11, 7)
+languages:[en]
 nums.map(_ * 2)         // List(10, 2, 6, 22, 14)
+languages:[en]
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/fp-functions-are-values.md:97]: what's happening here?

File: _zh-cn/overviews/scala3-book/scala-features.md [hunk 3]
```diff
@@ -192,29 +112,24 @@ val p = Person("Martin", "Odersky")
...
 {% tab 'Scala 2 and 3' for=scala-features-4 %}
 ```scala
+languages:[en]
 nums.map(i => i * 2)   // long form
+languages:[en]
 nums.map(_ * 2)        // short form
 
+languages:[en]
 nums.filter(i => i > 1)
+languages:[en]
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/scala-features.md:127]: why are these in a code block?

File: _zh-cn/overviews/scala3-book/first-look-at-types.md (hunk 1 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/first-look-at-types.md]: done

File: _zh-cn/overviews/scala3-book/fp-functions-are-values.md (hunk 2 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/fp-functions-are-values.md]: deleted. I used wrong regex in emacs.

File: _zh-cn/overviews/scala3-book/scala-features.md (hunk 3 above)
Comment (benluo) [reply _zh-cn/overviews/scala3-book/scala-features.md]: deleted. I used wrong regex in emacs.

File: _zh-cn/overviews/scala3-book/types-type-classes.md [hunk 4]
```diff
@@ -1,57 +0,0 @@
...
-previous-page: ca-given-using-clauses
-next-page: ca-context-bounds
-
-partof: scala3-book
-overview-name: "Scala 3 — Book"
-layout: multipage-overview
-permalink: "/zh-cn/scala3/book/:title.html"
----
-
-A _type class_ is an abstract, parameterized type that lets you add new behavior to any closed data type without using sub-typing.
```
Comment (bishabosha) [inline _zh-cn/overviews/scala3-book/types-type-classes.md:15]: Why did you delete this file? - it causes a link to break in `scala-for-python-devs.md`

Comment (bishabosha) [reply _zh-cn/overviews/scala3-book/types-type-classes.md:15]: ok I see, its in English

Comment (bishabosha) [reply _zh-cn/overviews/scala3-book/types-type-classes.md:15]: you will need to comment out links to this page in `scala-for-python-devs.md`

**Review by bishabosha — APPROVED** [review]
